src/DefaultBuilder/src/StaticWebAssetsLoader.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore
+{
+    internal class StaticWebAssetsLoader
+    {
+        internal const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+
+        internal static void UseStaticWebAssets(IWebHostEnvironment environment)
+        {
+            using (var manifest = ResolveManifest(environment))
+            {
+                if (manifest != null)
+                {
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var additionalFiles = StaticWebAssetsReader.Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (additionalFiles.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                additionalFiles.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(additionalFiles);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            var assembly = Assembly.Load(environment.ApplicationName);
```

Comment:
This is a poor assumption that the app name is an assembly. It could be anything.

This needs error handling.

Reply:
This is going away as we are moving the manifest to be a file, but this is also how MVC finds and loads the entry point, and I don't remember getting complaints about it.

Reply:
Added a try-catch and silently ignore if there is an error.